consensus-types/blocks/setters.go

```diff
@@ -132,3 +133,21 @@ func (b *SignedBeaconBlock) SetBlobKzgCommitments(c [][]byte) error {
 		return errIncorrectBlockVersion
 	}
 }
+
+// SetPayloadAttestations sets the payload attestations in the block.
+func (b *SignedBeaconBlock) SetPayloadAttestations(p []*eth.PayloadAttestation) error {
+	if b.version < version.EPBS {
+		return consensus_types.ErrNotSupported("PayloadAttestations", b.version)
+	}
+	b.block.body.payloadAttestations = p
```

Comment:
Do we need to check the length of `p` here? or this is enforced in the pipeline.

Reply:
I think it should be enforced by the caller (in the pipeline), and the function doesn't care, similar to the rest of the setters. I'm happy to add a check if you think it should be here.